File: 2025-10-03 REFATORACAO (Java)/classes/Distrito.java
package JogosVorazes.classes;

public class Distrito {
    private String nome;
    private String[] recursos;
    private int populacao;

    public Distrito(String nome, String[] recursos, int populacao) {
        this.nome = nome;
        this.recursos = recursos;
        this.populacao = populacao;
    }

    public String getNome() {
        return nome;
    }

    public void setNome(String nome) {
        this.nome = nome;
    }

    public String[] getRecursos() {
        return recursos;
    }

    public void setRecursos(String[] recursos) {
        this.recursos = recursos;
    }

    public int getPopulacao() {
        return populacao;
    }

    public void setPopulacao(int populacao) {
        this.populacao = populacao;
    }
}

File: 2025-10-03 REFATORACAO/distrito.ts
export class Distrito {
    private nome: string;
    private recursos: string[];
    private populacao: number;

constructor(nome: string, recursos: string[], populacao: number) {
    this.nome = nome;
    this.recursos = recursos;
    this.populacao = populacao;
}

public setNome(nomeRecebido: string): void {
    this.nome = nomeRecebido;
}

public setRecursos(recursosRecebidos: string[]): void {
    this.recursos = recursosRecebidos;
}

public setPopulacao(populacaoRecebida: number): void {
    this.populacao = populacaoRecebida;
}

public getNome(): string {
    return this.nome;
}

public getRecursos(): string[] {
    return this.recursos;
}

public getPopulacao(): number {
    return this.populacao;
}

}